frontend/src/components/ViewActivities/activitylist.js

```diff
@@ -8,17 +8,32 @@ import app from '../Firebase';
 const db = app.firestore();
 
 /**
- * Gets the list of activities from the server. 
+ * ReactJS class for the list of activities. 
  * 
- * @param {string} tripId The trip ID.
+ * @property {Object} props ReactJS props.
+ * @property {string} tripId The tripID.
  */
-export async function getActivityList(tripId) {
-  return new Promise(function(resolve, reject) {
-    let tripActivities = [];
+class ActivityList extends React.Component {
+  /** @inheritdoc */
+  constructor(props) {
+    super(props);
+    this.state = { days : [] };
+
+    this.getActivityList = this.getActivityList.bind(this);
+  }
+
+  /**
+   * Gets the list of activities from the server. 
```

Comment:
You may want to mention what state you set in here. That is up to you though so don't have to add if you don't think it is important.